src/parser.rs

```diff
@@ -241,29 +243,31 @@ pub fn parse_sof<R: Read>(reader: &mut R, marker: Marker) -> Result<FrameInfo> {
 }
 
 /// Returns ceil(x/y), requires x>0
-fn ceil_div(x: u32, y: u32) -> u16 {
-    assert!(x>0 && y>0, "invalid dimensions");
-    (1 + ((x - 1) / y)) as u16
+fn ceil_div(x: u32, y: u32) -> Result<u16> {
+    if x ==0 || y == 0 {
+        return Err(Error::Format("invalid dimensions".to_owned()));
```

Comment:
When I opened the issue report, I was unsure how the code even reached this point in the first place. At least I was under the impression that both numbers were checked individually before. It doesn't really matter. However, would you add a `TODO` to validate all inputs more closely such that we might error earlier?

Reply:
Sure, will do!